sdk/cosmos/azure-cosmos/azure/cosmos/_availability_strategy_handler.py

```diff
@@ -0,0 +1,319 @@
+
+    def _get_applicable_endpoints(self, request: RequestObject, global_endpoint_manager: Any) -> List[str]:
+        """Get list of applicable endpoints for hedging based on operation type.
+
+        :param request: Request object containing operation type and other parameters
+        :type request: RequestObject
+        :param global_endpoint_manager: Manager for endpoint routing and availability
+        :type global_endpoint_manager: Any
+        :returns: List of region names that can be used for hedging
+        :rtype: List[str]
+        """
+        applicable_endpoints = []
+        if _OperationType.IsWriteOperation(request.operation_type):
+            regional_context_list = global_endpoint_manager.get_applicable_write_regional_routing_contexts(request)
+        else:
+            regional_context_list = global_endpoint_manager.get_applicable_read_regional_routing_contexts(request)
+
+        if regional_context_list is not None:
+            for regional_context in regional_context_list:
+                applicable_endpoints.append(
+                    global_endpoint_manager.get_region_name(
+                        regional_context.get_primary(),
+                        _OperationType.IsWriteOperation(request.operation_type)
+                    )
+                )
+
+        return applicable_endpoints
+
+# Global handler instance
+_cross_region_hedging_handler = CrossRegionHedgingHandler()
+
+def execute_with_hedging(
+    request_params: RequestObject,
+    global_endpoint_manager: _GlobalPartitionEndpointManagerForCircuitBreaker,
+    request: HttpRequest,
+    execute_request_fn: Callable[..., ResponseType]
+) -> ResponseType:
+    """Execute a request with hedging based on the availability strategy.
+
+    :param request_params: Parameters for the request including operation type and strategy
+    :type request_params: RequestObject
+    :param global_endpoint_manager: Manager for handling global endpoints and circuit breaking
+    :type global_endpoint_manager: _GlobalPartitionEndpointManagerForCircuitBreaker
+    :param request: The HTTP request to be executed
+    :type request: HttpRequest
+    :param execute_request_fn: Function to execute the actual request
+    :type execute_request_fn: Callable[..., ResponseType]
+    :returns: A tuple containing the response data and headers
+    :rtype: Tuple[Dict[str, Any], Dict[str, Any]]
+    :raises: Any exceptions raised by the hedging handler's execute_request method
+    """
+    availability_strategy = request_params.availability_strategy
+    if isinstance(availability_strategy, CrossRegionHedgingStrategy):
+        return _cross_region_hedging_handler.execute_request(
+            request_params,
+            global_endpoint_manager,
+            request,
+            execute_request_fn
+        )
+    raise ValueError("Unsupported availability strategy type: {0}".format(CrossRegionHedgingStrategy.__class__))
```

Comment:
Not everything has a `__class__` attr
```suggestion
    raise ValueError(f"Unsupported availability strategy type: {type(CrossRegionHedgingStrategy)}")
```

Reply:
thanks, updated